qoffer-infrastructure/src/main/groovy/life/qbic/portal/qoffer2/customers/CustomerDbConnector.groovy

```diff
@@ -81,23 +152,122 @@ class CustomerDbConnector implements CreateCustomerDataSource, UpdateCustomerDat
   @Override
   void addCustomer(Customer customer) throws DatabaseQueryException {
     try {
-      databaseQueries.addCustomer(customer)
-    } catch (DatabaseQueryException e) {
-      throw new DatabaseQueryException(e.message)
+      if (customerExists(customer)) {
+        throw new DatabaseQueryException("Customer is already in the database.")
+      }
+      Connection connection = connectionProvider.connect()
+      connection.setAutoCommit(false)
+
+      connection.withCloseable {it ->
+        try {
+          int customerId = createNewCustomer(it, customer)
+          storeAffiliation(it, customerId, customer.affiliations)
+          connection.commit()
+        } catch (Exception e) {
+          log.error(e.message)
+          log.error(e.stackTrace.join("\n"))
+          connection.rollback()
+          connection.close()
+          throw new DatabaseQueryException("Could not create customer.")
```

Comment:
1. Can be put into a small method `tryCreateNewCustomer`, to make this method less verbose.

2. No need to close the connection, which is guaranteed by the implementation of the AutoClosable interface

Reply:
I will not address this in this PR since this PR only moves existing code into the DBConnector. But yes that was the idea.

Reply:
Do we want to track this in an issue?